### PR TITLE
chore: migrate existing geth + dek accounts to the Keychain based wallet

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

Builds on top of #2519 to add backward compatibility for existing user accounts.

This transparently migrates the existing geth + dek accounts into the Keychain based wallet.

The geth account is re-imported from the mnemonic the next time the user enters their PIN to perform an action that requires it (signing, updating their PIN, etc).

The dek is re-imported on the first launch if a user account is already set. It doesn't require the PIN (by design).

### Tested

- Updated unit tests for the KeychainWallet
- Manually upgrading the app from the the geth based wallet to the new Keychain based wallet on both iOS and Android.

### How others should test

1. Have an existing app <= v1.33 with a user account set
2. Ensure TXs can be sent and and the seed phrase can be seen.
3. Update the app with version >= 1.34 (containing this commit)
4. Make sure all things related to signing work as expected
    - sending transaction
    - swaping
    - signing with dapps
    - phone number verification
    - updating the PIN

### Related issues

- Part of #2392 

### Backwards compatibility

Yes

### CODE REVIEW

File: src/web3/KeychainWallet.test.ts [hunk 1]
```diff
@@ -366,4 +378,294 @@ describe('KeychainWallet', () => {
       })
     })
   })
+
+  // This tests migration from a Geth KeyStore account
+  describe('migration from an existing geth account', () => {
+    // @ts-expect-error
```
Comment (kathaypacific) [inline src/web3/KeychainWallet.test.ts:384]: are these variables used?

File: src/web3/KeychainWallet.test.ts [hunk 2]
```diff
@@ -366,4 +378,294 @@ describe('KeychainWallet', () => {
...
+
+      it('directly reads the private key from the keychain when unlocking', async () => {
+        await expect(
+          wallet.unlockAccount(GETH_ACCOUNT_ADDRESS, 'password', UNLOCK_DURATION)
+        ).resolves.toBe(true)
+
+        expect(Keychain.setGenericPassword).toHaveBeenCalledTimes(0)
+      })
+    })
+  })
```
Comment (kathaypacific) [inline src/web3/KeychainWallet.test.ts:670]: nice tests!!

**Review by kathaypacific — APPROVED** [review]
super 🆒.  !!!

File: src/web3/KeychainWallet.ts [hunk 3]
```diff
@@ -3,20 +3,24 @@ import { UnlockableWallet } from '@celo/wallet-base'
...
  * A wallet which uses the OS keychain to store private keys
  */
 export class KeychainWallet extends RemoteWallet<KeychainSigner> implements UnlockableWallet {
-  constructor() {
+  /**
+   * Construct a new instance of the Keychain wallet
+   * @param importMnemonicAccount ImportMnemonicAccount the existing account to import from the mnemonic, if not already present in the keychain
+   */
+  constructor(protected importMnemonicAccount: ImportMnemonicAccount) {
     super()
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply src/web3/KeychainWallet.ts:19]: This is a shortcut syntax to assign the parameter to a instance property.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on iOS TestFlight Release build V1.34.0 & updated status in test rails.

**Observation :**
- User is able to perform transaction from previous builds and also able to upgrade to latest build
**Note :** For Android we are blocking the upgrade test case as we are not able to update it due to work profile issue 
- User is able to send request and complete the transaction by sending the money
- User is able to connect to Dapps and swap the funds
- User is able to confirm Phone number via create and restore account flow
- User is able to update the PIN to latest PIN

**Test rail links:**
**iOS 14:** https://valoraapp.testrail.io/index.php?/tests/view/41889

**Devices:** iPhone 12(14.7.1)

CC: @jeanregisser 

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on iOS Test Flight Release build V1.34.0 & updated status in test rails.

**Observation :**

- User is able to perform transaction from previous builds and also able to upgrade to latest build
- Note : For Android we are blocking the upgrade test case as we are not able to update it due to work profile issue
- User is able to send request and complete the transaction by sending the money
- User is able to connect to Dapps and swap the funds
- User is able to confirm Phone number via create and restore account flow
- User is able to update the PIN to latest PIN

**Test rail links:**
**iOS 15:** https://valoraapp.testrail.io/index.php?/tests/view/41920

**Devices:** iPhone 13 mini(15.1.1)

**CC:** @jeanregisser 

Thanks..!!

**Pull request merged**